test/Test/Toml/Parser/Integer.hs

```diff
@@ -81,3 +81,27 @@ integerSpecs = describe "integerP" $ do
         it "can parse numbers when hex digits are in both lowercase and uppercase" $ do
             parseInteger "0xAbCdEf" 0xAbCdEf
             parseInteger "0xaBcDeF" 0xaBcDeF
+        it "can parse numbers with underscore in hexadecimal representation" $ do
+            parseInteger "0xAb_Cd_Ef" 0xabcdef
+            parseInteger "0xA_bcd_ef" 0xabcdef
+            parseInteger "0x123_abc" 0x123abc
+            parseInteger "0xa_b_c_1_2_3" 0xabc123
+        it "can't parse when underscore is between hexadecimal prefix and suffix" $ do
+            parseInteger "0x_Abab_ca" 0
+            parseInteger "0x_ababbac" 0
```

Comment:
I believe these tests should fail, not return `0`

Reply:
You are right.